File: backend/models/cart_item.go
package models

import "time"

type CartItem struct {
	ID        uint      `json:"id" gorm:"primaryKey"`
	CreatedAt time.Time `json:"created_at"`
	UpdatedAt time.Time `json:"updated_at"`
	CartID    uint      `json:"cart_id"`
	ItemID    uint      `json:"item_id"`
	Item      Item      `json:"item" gorm:"foreignKey:ItemID"`
	Quantity  int       `json:"quantity"`
}

File: frontend/types/cart_item.ts
import { z } from 'zod';
import { ItemSchema } from './item';

export const CartItemSchema = z.object({
  id: z.number(),
  created_at: z.string().datetime().optional(),
  updated_at: z.string().datetime().optional(),
  cart_id: z.number(),
  item_id: z.number(),
  item: ItemSchema.optional(),
  quantity: z.number(),
});

export const CartDetailSchema = z.object({
  id: z.number(),
  created_at: z.string().datetime().optional(),
  updated_at: z.string().datetime().optional(),
  cart_id: z.number(),
  item_id: z.number(),
  item: ItemSchema.optional(),
  quantity: z.number(),
});

export type CartItem = z.infer<typeof CartItemSchema>;